overlord/hookstate/ctlcmd/ctlcmd.go

```diff
@@ -91,19 +91,44 @@ func addCommand(name, shortHelp, longHelp string, generator func() command) {
 	}
 }
 
+type ForbiddenCommand struct {
+	Uid     uint32
+	Command string
+}
+
+func (f ForbiddenCommand) Error() string {
+	return fmt.Sprintf("cannot use %q with uid %d, try with sudo", f.Command, f.Uid)
+}
+
+func (f *ForbiddenCommand) Execute(args []string) error {
+	return f
+}
+
+func (f *ForbiddenCommand) setStdout(w io.Writer)                 {}
+func (f *ForbiddenCommand) setStderr(w io.Writer)                 {}
+func (f *ForbiddenCommand) setContext(context *hookstate.Context) {}
+func (f *ForbiddenCommand) context() *hookstate.Context {
```

Comment:
None of these methods seem necessary for the actual functionality we need for this feature. The fact they are all empty also backs that idea. See below for the alternative.

Reply:
Indeed, great idea, fixed per your suggestion, thanks!